### PR TITLE
open files only if recording is enabled

### PR DESCRIPTION
**Pull request by koldn**

### CODE REVIEW

**Comment by MentionBot-JSFlight** [comment]
@koldn, thanks for your PR! By analyzing the history of the files in this pull request, we identified @d0k1 to be a potential reviewer.

File: recorder/src/main/java/com/focusit/jsflight/recorder/internalevent/httprequest/HttpRecorderFilterBase.java [hunk 1]
```diff
@@ -14,12 +14,17 @@
 {
     public static final String ALREADY_FILTERED_SUFFIX = ".FILTERED";
     private static final AtomicBoolean ENABLED = new AtomicBoolean(false);
+    private static final AtomicBoolean NEED_OPEN_OUTPUT = new AtomicBoolean(false);
```
Comment (d0k1) [inline recorder/src/main/java/com/focusit/jsflight/recorder/internalevent/httprequest/HttpRecorderFilterBase.java]: is a situation possible when someone has enabled = true and need_open_output = false?
As for me, there should be one variable to determine if output file should be opened or not.

**Comment by d0k1** [comment]
@koldn AR

**Comment by d0k1** [comment]
Feel free to merge it

**Pull request merged**